daemon/create.go

```diff
@@ -95,7 +95,14 @@ func (daemon *Daemon) create(params types.ContainerCreateConfig, managed bool) (
 		if err != nil {
 			return nil, err
 		}
-		os = img.OS
+		if img.OS != "" {
+			os = img.OS
+		} else {
+			// default to the host OS except on Windows with LCOW
+			if runtime.GOOS == "windows" && system.LCOWSupported() {
+				os = "linux"
+			}
+		}
 		imgID = img.ID()
```

Comment:
We should also validate that the os in image is actually valid, and produces a correct error if it is not instead of letting it panic. This can be follow-up in another PR as it is related but not quite the same issue and there are probably more places that need this check.

Reply:
This is indirectly addressed by the store re-coalescing PR.